lib/puppet/parser/functions/enclose_ipv6.rb

```diff
@@ -28,16 +28,18 @@ module Puppet::Parser::Functions
     result = []
 
     input.each do |val|
-      begin
-        ip = IPAddr.new(val)
-      rescue *rescuable_exceptions
-        raise(Puppet::ParseError, "enclose_ipv6(): Wrong argument "+
-          "given #{val} is not an ip address.")
+      unless val == '*'
+        begin
+          ip = IPAddr.new(val)
+        rescue *rescuable_exceptions
+          raise(Puppet::ParseError, "enclose_ipv6(): Wrong argument "+
+                "given #{val} is not an ip address.")
+        end
+        val = "[#{ip.to_s}]" if ip.ipv6?
       end
-      val = "[#{ip.to_s}]" if ip.ipv6?
-      result = [result,val]
+      result << val
     end
 
-    return result.flatten.compact
```

Comment:
It's because I changed ```result = [result,val]``` to ```result << val``` which makes more sense. ```input``` is already flattened in line 27, so we do not need to flatten it once again.